File: recruitment-service/src/main/java/pe/albrugroup/recruitment_service/entity/response/GrupoCapacitacionDetalleResponse.java
package pe.albrugroup.recruitment_service.entity.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import pe.albrugroup.recruitment_service.entity.enums.EstadoCapacitacionPostulante;

import java.time.Instant;
import java.time.LocalDate;

@Builder @Getter @Setter
@AllArgsConstructor @NoArgsConstructor
public class GrupoCapacitacionDetalleResponse {

    private Long id;
    private EstadoCapacitacionPostulante estadoCapacitacion;
    private LocalDate fechaAsignacion;
    private LocalDate fechaResultado;
    private Long idEmpleadoContratado;
    private LocalDate fechaContratacion;
    private Boolean cumplioTresMeses;
    private LocalDate fechaCumplioTresMeses;
    private Instant createdAt;
    private Instant updatedAt;
    private PostulacionResponse postulacion;
}

File: frontend/src/app/shared/models/recruitment/grupo-capacitacion-detalle-response.ts
import { PostulacionResponse } from './postulacion-response';

export interface GrupoCapacitacionDetalleResponse {
  id: number;
  estadoCapacitacion: string;
  fechaAsignacion: string;
  fechaResultado: string | null;
  idEmpleadoContratado: number | null;
  fechaContratacion: string | null;
  cumplioTresMeses: boolean | null;
  fechaCumplioTresMeses: string | null;
  createdAt: string;
  updatedAt: string | null;
  postulacion: PostulacionResponse;
}
